File: server/src/main/java/com/manders/ecommerce/entity/Product.java
package com.manders.ecommerce.entity;

import java.util.Date;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "product")
public class Product {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id")
  private Long id;

  @Column(name = "sku")
  private String sku;
  
  @Column(name = "name")
  private String name;
  
  @Column(name = "description")
  private String description;
  
  @Column(name = "unit_price")
  private int unitPrice;
  
  @Column(name = "image_url")
  private String imgUrl;

  @Column(name = "active")
  private boolean active;
  
  @Column(name = "units_in_stock")
  private int unitsInStock;
  
  @Column(name = "date_created")
  @CreationTimestamp
  private Date dateCreated;
  
  @Column(name = "last_updated")
  @UpdateTimestamp
  private Date lastUpdated;
  
  @ManyToOne
  @JoinColumn(name = "category_id", nullable = false)
  private ProductCategory category;
}

File: client/src/app/common/product.ts
export class Product {
    public id!: number;
    public sku!: string;
    public name!: string;
    public description!: string;
    public unitPrice!: number;
    public imageUrl!: string;
    public active!: boolean;
    public unitsInStock!: number;
    public dateCreated!: Date;
    public lastUpdated!: Date;
    constructor(
        id?: number,
        sku?: string,
        name?: string,
        description?: string,
        unitPrice?: number,
        imageUrl?: string,
        active?: boolean,
        unitsInStock?: number,
        dateCreated?: Date,
        lastUpdated?: Date
    ) {

    }
}
